scheduler/pkg/store/utils/validate_test.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestCheckName(t *testing.T) {
+	g := NewGomegaWithT(t)
+	tests := []struct {
+		name           string
+		value          string
+		expectedResult bool
+	}{
+		{
+			"a valid name",
+			"this-NAME_is_valid1234-5",
+			true,
+		},
+		{
+			"a valid numerical name",
+			"111111111111111111",
+			true,
+		},
+		{
+			"an invalid name that begins and ends with something alphanumeric",
+			"1-----______A",
+			true,
+		},
+
+		{
+			"an invalid name that doesn't begin and end with something alphanumeric",
+			"-----______",
+			false,
+		},
+		{
+			"an invalid name that doesn't end with something alphanumeric",
+			"A-----______",
+			false,
+		},
+		{
+			"an invalid name that doesn't begin with something alphanumeric",
+			"-----______1",
+			false,
+		},
+		{
+			"an invalid name with a dot",
+			"this-Is-some-NA.ME_that_is_valid1234-5",
+			false,
+		},
+	}
```

Comment:
Suggest to add a test-case for uppercase letters failing the validation

Reply:
Added it.